client/v2/autocli/flag/field.go

```diff
@@ -81,7 +86,61 @@ func (b *Builder) resolveFlagType(field protoreflect.FieldDescriptor) Type {
 		if typ != nil {
 			return compositeListType{simpleType: typ}
 		}
+		return nil
+	}
+	if field.IsMap() {
+		keyKind := field.MapKey().Kind()
+		valType := b.resolveFlagType(field.MapValue())
+		if valType != nil {
+			switch keyKind {
+			case protoreflect.StringKind:
+				ct := new(compositeMapType[string])
+				ct.keyValueResolver = func(s string) (string, error) { return s, nil }
+				ct.valueType = valType
+				ct.keyType = "string"
+				return ct
+			case protoreflect.Int32Kind, protoreflect.Sint32Kind, protoreflect.Sfixed32Kind:
+				ct := new(compositeMapType[int32])
+				ct.keyValueResolver = func(s string) (int32, error) {
+					i, err := strconv.ParseInt(s, 10, 32)
+					return int32(i), err
+				}
+				ct.valueType = valType
+				ct.keyType = "int32"
+			case protoreflect.Int64Kind, protoreflect.Sint64Kind, protoreflect.Sfixed64Kind:
+				ct := new(compositeMapType[int64])
+				ct.keyValueResolver = func(s string) (int64, error) {
+					i, err := strconv.ParseInt(s, 10, 64)
+					return i, err
+				}
+				ct.valueType = valType
+				ct.keyType = "int64"
+			case protoreflect.Uint32Kind, protoreflect.Fixed32Kind:
+				ct := new(compositeMapType[uint32])
+				ct.keyValueResolver = func(s string) (uint32, error) {
+					i, err := strconv.ParseUint(s, 10, 32)
+					return uint32(i), err
+				}
+				ct.valueType = valType
+				ct.keyType = "uint32"
+			case protoreflect.Uint64Kind, protoreflect.Fixed64Kind:
+				ct := new(compositeMapType[uint64])
+				ct.keyValueResolver = func(s string) (uint64, error) {
+					i, err := strconv.ParseUint(s, 10, 64)
+					return i, err
+				}
+				ct.valueType = valType
+				ct.keyType = "uint64"
+			case protoreflect.BoolKind:
+				ct := new(compositeMapType[bool])
+				ct.keyValueResolver = strconv.ParseBool
+				ct.valueType = valType
+				ct.keyType = "bool"
```

Comment:
All these `ct` values are freshly created in `ct := new(compositeMapType[T])` but never returned hence never used and the compiler ideally should reject compiling this code. Please make an update to return each `ct` or otherwise just delete that dead code.